src/main/java/com/exadel/discountwebapp/user/entity/User.java

```diff
@@ -57,9 +61,9 @@
     @JoinColumn(name = "rol_id")
     private Role role;
 
-    @OneToMany(mappedBy = "user")
+    @ManyToMany(mappedBy = "users")
     @EqualsAndHashCode.Exclude
-    private List<UserDiscount> discounts = new ArrayList<>();
+    private List<Discount> discounts = new ArrayList<>();
```

Comment:
why did you change this?

Reply:
Searching with specification doesnt work without this changes 

/api/discounts?query=users.id:1
Or 
/api/users?query=discounts.id:3

Reply:
/api/discounts?query=users.user.id:1 search discounts by user id
/api/discounts?query=users.id.userId:1 search discounts by user id